app/src/main/java/com/infomaniak/mail/ui/main/menu/MoveFragment.kt

```diff
@@ -104,4 +129,40 @@ class MoveFragment : MenuFoldersFragment() {
             },
         )
     }
+
+    private fun setSearchBarUi(allFolders: List<Folder>, currentFolderId: String) = with(binding) {
+        searchResultsAdpater = FolderAdapter(
+            isInMenuDrawer,
+            currentFolderId,
+            shouldIndent = false,
+            onFolderClicked = ::onFolderSelected,
+        )
+        searchResultsList.adapter = searchResultsAdpater
+
+        searchInputLayout.setOnClearTextClickListener { trackMoveSearchEvent(SEARCH_DELETE_NAME) }
+        searchTextInput.apply {
+            doOnTextChanged { text, _, _, _ ->
+                toggleFolderListsVisibility(!text.isNullOrBlank())
+                if (text?.isNotBlank() == true) moveViewModel.filterFolders(text.toString(), allFolders, shouldDebounce = true)
+                if (!hasAlreadyTrackedFolderSearch) {
+                    trackMoveSearchEvent("executeSearch")
+                    hasAlreadyTrackedFolderSearch = true
+                }
+            }
+
+            handleEditorSearchAction { query ->
+                moveViewModel.filterFolders(query, allFolders, shouldDebounce = false)
+                trackMoveSearchEvent(SEARCH_VALIDATE_NAME)
+            }
+        }
+    }
+
+    private fun toggleFolderListsVisibility(isSearching: Boolean) = with(binding) {
+        searchResultsList.isVisible = isSearching
+        moveFoldersLists.isGone = isSearching
+    }
+
+    private fun observeSearchResults() {
+        moveViewModel.filterResults.observe(viewLifecycleOwner) { folders -> searchResultsAdpater.setFolders(folders) }
```

Comment:
at recreation sa crash if we had a search because the `lateinit` is initialized much later